src/user-lists/user-list-stat.service.ts

```diff
@@ -0,0 +1,337 @@
+      .createQueryBuilder()
+      .select("split_part(repos.full_name, '/', 1)", "orgId")
+      .addSelect("split_part(repos.full_name, '/', 2)", "projectId")
+      .addSelect("COUNT(pr.id)", "contributions")
+
+      // grab pull requests first
+      .from("pull_requests", "pr")
+
+      // and join them with repos
+      .innerJoin("repos", "repos", 'pr."repo_id" = "repos"."id"')
+
+      // join with filtered users from the user list
+      .innerJoin(
+        (subQuery) =>
+          subQuery
+            .select("users.login", "user_login")
+            .from("user_list_contributors", "user_list_contributors")
+            .innerJoin("users", "users", '"user_list_contributors"."user_id" = "users"."id"')
+            .where(`"user_list_contributors"."list_id" = '${listId}'`),
+        "filtered_users",
+        'pr."author_login" = "filtered_users"."user_login"'
+      )
+
+      .where(`pr."updated_at" BETWEEN NOW() - INTERVAL '${range} days' AND NOW()`)
+      .groupBy("repos.full_name");
+
+    const entities: DbContributionsProjects[] = await queryBuilder.getRawMany();
+
+    return entities;
+  }
+
+  private applyActiveContributorsFilter(
+    queryBuilder: SelectQueryBuilder<DbUserListContributor>,
+    startDate: string,
+    range = 30
+  ) {
+    queryBuilder
+      .leftJoin(
+        `(
+        SELECT DISTINCT "author_login"
+        FROM "pull_requests"
+        WHERE "pull_requests"."updated_at" BETWEEN '${startDate}'::DATE - INTERVAL '${range} days'
+          AND '${startDate}'::DATE
+      )`,
+        "current_month_prs",
+        `"users"."login" = "current_month_prs"."author_login"`
+      )
+      .leftJoin(
+        `(
+        SELECT DISTINCT "author_login"
+        FROM "pull_requests"
+        WHERE "pull_requests"."updated_at" BETWEEN '${startDate}'::DATE - INTERVAL '${range + range} days'
+          AND '${startDate}'::DATE - INTERVAL '${range} days'
+      )`,
+        "previous_month_prs",
+        `"users"."login" = "previous_month_prs"."author_login"`
+      )
+      .where(`"previous_month_prs"."author_login" IS NOT NULL`)
+      .andWhere(`"current_month_prs"."author_login" IS NOT NULL`);
+  }
+
+  private applyNewContributorsFilter(
+    queryBuilder: SelectQueryBuilder<DbUserListContributor>,
+    startDate: string,
+    range = 30
+  ) {
+    queryBuilder
+      .leftJoin(
+        `(
+            SELECT DISTINCT "author_login"
+            FROM "pull_requests"
+            WHERE "pull_requests"."updated_at" BETWEEN NOW() - INTERVAL '${range} days'
+              AND '${startDate}'::DATE
+          )`,
+        "current_month_prs",
+        `"users"."login" = "current_month_prs"."author_login"`
+      )
+      .leftJoin(
+        `(
+            SELECT DISTINCT "author_login"
+            FROM "pull_requests"
+            WHERE "pull_requests"."updated_at" BETWEEN NOW() - INTERVAL '${range + range} days'
+              AND '${startDate}'::DATE - INTERVAL "${range} days"
```

Comment:
Ran into this from running locally

```suggestion
              AND '${startDate}'::DATE - INTERVAL '${range} days'
```